src/backend/BackendInterface.h

```diff
@@ -411,52 +529,94 @@ class BackendInterface
-    /// references.
+    /**
+     * @brief Write a new NFT.
+     *
+     * @param data Passed in as an r-value reference.
+     */
     virtual void
     writeNFTs(std::vector<NFTsData>&& data) = 0;
 
+    /**
+     * @brief Write a new set of account transactions.
+     *
+     * @param data Passed in as an r-value reference.
+     */
     virtual void
     writeAccountTransactions(std::vector<AccountTransactionsData>&& data) = 0;
 
+    /**
+     * @brief Write a new transaction for a specific NFT.
+     *
+     * @param data Passed in as an r-value reference.
+     */
     virtual void
     writeNFTTransactions(std::vector<NFTTransactionsData>&& data) = 0;
 
+    /**
+     * @brief Write a new successor.
+     *
+     * @param key Passed in as an r-value reference.
+     * @param seq Unsigned 32-bit integer.
+     * @param successor Passed in as an r-value reference.
+     */
     virtual void
     writeSuccessor(
         std::string&& key,
         std::uint32_t const seq,
         std::string&& successor) = 0;
 
-    /// Tell the database we are about to begin writing data for a particular
-    /// ledger.
+    /*! @brief Tells database we are going to write data for a specific ledger.
+     */
     virtual void
     startWrites() const = 0;
 
-    /// Tell the database we have finished writing all data for a particular
-    /// ledger
-    /// TODO: change the return value to represent different results. committed,
-    /// write conflict, errored, successful but not committed
+    /**
+     * @brief Tells database we finished writing all data for a specific ledger.
+     *
+     * TODO: change the return value to represent different results:
+     * Committed, write conflict, errored, successful but not committed
+     *
+     * @param ledgerSequence
+     * @return true
+     * @return false
+     */
     bool
     finishWrites(std::uint32_t const ledgerSequence);
 
+    /**
+     * @brief Selectively delets parts of the database.
+     *
+     * @param numLedgersToKeep
+     * @param yield
+     * @return true
+     * @return false
+     */
     virtual bool
     doOnlineDelete(
         std::uint32_t numLedgersToKeep,
         boost::asio::yield_context& yield) const = 0;
 
-    /// Open the database. Set up all of the necessary objects and
-    /// datastructures. After this call completes, the database is ready for
-    /// use.
+    /**
+     * @brief Opens the database.
+     *
+     * Open the database. Set up all of the necessary objects and
+     * datastructures. After this call completes, the database is
+     * ready for use.
+     *
+     * @param readOnly
+     */
     virtual void
     open(bool readOnly) = 0;
 
-    // Close the database, releasing any resources
+    /*! @brief Closes the database, releasing any resources. */
     virtual void
     close(){};
 
-    // *** private helper methods
+    /**
+     * @brief Private helper methods.
```

Comment:
i'm not sure this will come out as you expect in the doc. you need to verify once you fix the doc output to actually contain this file's documentation.